sorbet/rbi/shims/set.rbi

```diff
@@ -1,5 +1,6 @@
 # typed: strict
 
 class Set
+  sig { returns(T.self_type) }
```

Comment:
Should we upstream this change? https://github.com/sorbet/sorbet/blob/master/rbi/stdlib/set.rbi#L174-L178